### PR TITLE
Add LAPI and agent timing prom metrics

### LINKED ISSUE #1472: Improvement/Metrics : Extend Local API metrics
Currently, the prometheus metrics provided by the API are very scarce, which can make some corner case debugging non-trivial.
Opening this issue to track the metrics we might want to add :

 - last X requests response time (especially for bouncers)
 - number of IPs distributed to bouncers ?

### PR DESCRIPTION
**Pull request by buixor**
While looking at #1472 and the past issues of v1.3.2, measuring response time seemed relevant :)

This PR adds the following metrics:
  - [x] `cs_lapi_request_duration_seconds`: histogram with response time of LAPI, with a label per endpoint and method
  - [x] `cs_active_decisions`: gauge of active decisions, with labels for scenario, origin and action
  - [x] `cs_parsing_time_seconds`: histogram of parsing time
  - [x] `cs_bucket_pour_seconds`: histogram of parsing time
  - [x] `cs_alerts`: gauge of total alerts in the DB excluding the alerts created by CAPI pull, with the reason as a label

Will close #1472.

### CODE REVIEW

File: cmd/crowdsec-cli/metrics.go [hunk 1]
```diff
@@ -329,6 +352,30 @@ func ShowPrometheus(url string) {
 			}
 		}
 
+		decisionsTable := tablewriter.NewWriter(os.Stdout)
+		decisionsTable.SetHeader([]string{"Scenario", "Origin", "Action", "Hits"})
```
Comment (buixor) [inline cmd/crowdsec-cli/metrics.go:356]: if it's a counter of active decisions, we might find something more appropriate than `Hits`. `Count` or `Current` or something around that maybe ?

File: cmd/crowdsec-cli/metrics.go [hunk 2]
```diff
@@ -329,6 +352,30 @@ func ShowPrometheus(url string) {
...
+					row = append(row, origin)
+					row = append(row, action)
+					row = append(row, fmt.Sprintf("%d", hits))
+					decisionsTable.Append(row)
+				}
+			}
+		}
+
+		alertsTable := tablewriter.NewWriter(os.Stdout)
+		alertsTable.SetHeader([]string{"Scenario", "Hits"})
```
Comment (buixor) [inline cmd/crowdsec-cli/metrics.go:371]: Same as above, maybe replace "Hits" by "Count" ?

File: cmd/crowdsec/metrics.go [hunk 3]
```diff
@@ -62,7 +63,82 @@ var globalCsInfo = prometheus.NewGauge(
...
+
+func computeDynamicMetrics(next http.Handler, dbClient *database.Client) http.HandlerFunc {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if dbClient == nil {
+			next.ServeHTTP(w, r)
+			return
+		}
+
+		decisionsFilters := make(map[string][]string, 0)
+		decisions, err := dbClient.QueryAllDecisionsWithFilters(decisionsFilters)
```
Comment (buixor) [inline cmd/crowdsec/metrics.go:108]: maybe a count would be better than a full select

File: cmd/crowdsec/metrics.go [hunk 4]
```diff
@@ -62,7 +63,82 @@ var globalCsInfo = prometheus.NewGauge(
...
+			globalActiveDecisions.With(prometheus.Labels{"scenario": d.Scenario, "origin": d.Origin, "action": d.Type}).Inc()
+		}
+
+		alertsFilter := map[string][]string{
+			"include_capi": {"false"},
+		}
+
+		globalAlerts.Reset()
+
+		alerts, err := dbClient.QueryAlertWithFilter(alertsFilter)
```
Comment (buixor) [inline cmd/crowdsec/metrics.go:125]: maybe a count would be better than a full select

File: cmd/crowdsec/metrics.go [hunk 5]
```diff
@@ -62,7 +63,82 @@ var globalCsInfo = prometheus.NewGauge(
...
+		Name: "cs_alerts",
+		Help: "Number of alerts (excluding CAPI).",
+	},
+	[]string{"scenario"},
+)
+
+var globalParsingHistogram = prometheus.NewHistogramVec(
+	prometheus.HistogramOpts{
+		Help:    "Time spent parsing a line",
+		Name:    "cs_parsing_time_seconds",
```
Comment (buixor) [inline cmd/crowdsec/metrics.go:85]: let's agree not to represent this in the console cli version

File: cmd/crowdsec/metrics.go [hunk 6]
```diff
@@ -62,7 +63,82 @@ var globalCsInfo = prometheus.NewGauge(
...
+		Buckets: []float64{0.0005, 0.001, 0.0015, 0.002, 0.0025, 0.003, 0.004, 0.005, 0.0075, 0.01},
+	},
+	[]string{"type", "source"},
+)
+
+var globalPourHistogram = prometheus.NewHistogramVec(
+	prometheus.HistogramOpts{
+		Name:    "cs_bucket_pour_seconds",
+		Help:    "Time spent pouring an event to buckets.",
+		Buckets: []float64{0.001, 0.002, 0.005, 0.01, 0.015, 0.02, 0.03, 0.04, 0.05},
```
Comment (buixor) [inline cmd/crowdsec/metrics.go:95]: let's agree not to represent this in the console cli version

**Review by buixor — COMMENTED** [review]
plz

**Review by nitescuc — APPROVED** [review]

**Review by buixor — COMMENTED** [review]
lgtm

**Pull request merged**